go/engine/merkle_audit.go

```diff
@@ -100,34 +101,39 @@ func (e *MerkleAudit) Shutdown() {
 	e.task.Shutdown()
 }
 
-// randSeqno picks a random number between [low, high).
-func randSeqno(lo keybase1.Seqno, hi keybase1.Seqno) (keybase1.Seqno, error) {
-	rangeBig := big.NewInt(int64(hi - lo))
-	n, err := rand.Int(rand.Reader, rangeBig)
-	if err != nil {
-		return keybase1.Seqno(0), err
+// randSeqno picks a random number between [low, high) that's different from prev.
+func randSeqno(lo keybase1.Seqno, hi keybase1.Seqno, prev *keybase1.Seqno) (keybase1.Seqno, error) {
+	for {
+		rangeBig := big.NewInt(int64(hi - lo))
+		n, err := rand.Int(rand.Reader, rangeBig)
+		if err != nil {
+			return keybase1.Seqno(0), err
+		}
+		newSeqno := keybase1.Seqno(n.Int64()) + lo
+		if prev == nil || *prev != newSeqno {
```

Comment:
Actually is it possible for this to loop forever?  What if the params are [0,1), with prev=0?

Reply:
Sure, but isn't that even less likely than the flake itself? Even locally / in dev mode we preload the database with quite a few roots.

Reply:
Still it makes me nervous, without any comment or anything.  Just make it error out in that case, I don't want to spin a goroutine up to 100% without knowing it sometime down the road.

Reply:
fixed